crowbar_framework/app/models/node_object.rb

```diff
@@ -259,6 +259,7 @@ def initialize(node)
     end
     # deep clone of @role.default_attributes, used when saving node
     @attrs_last_saved = @role.default_attributes.deep_dup
+    @notify_on_save = {}
```

Comment:
This variable really holds the notifications to be sent on the next save, if understand it correctly. Also looking at the point where the notifications are sent out, it seems, that we are not using the value in the hash, only the keys. I think a better name for this variable might help, and also making it an array/set would also help. I would name it something like: `on_save_events`. Thus the protocol would look like: the client adds/queues up items to the `on_save_events` collection, and those are being sent and flushed on save.